File: Swift/stringcalculator/NegativeNumberValidateRule.swift
//
//  NegativeNumberValidateRule.swift
//  stringcalculator
//
//  Created by Jorge Sánchez on 13/10/16.
//  Copyright © 2016 xurxodev. All rights reserved.
//

import Foundation

class NegativeNumberValidateRule: ValidateRule {
    func validate(number: Int) throws {
        if (number < 0){
            throw InvalidInputError.NegativeNumbersFound
        }
        
    }
}

File: Java/src/main/java/com/xurxodev/stringcalculator/rule/NegativeNumberValidateRule.java
package com.xurxodev.stringcalculator.rule;

import com.xurxodev.stringcalculator.NumberValidatorByValidateRules;
import com.xurxodev.stringcalculator.exception.NegativeNumbersNotAllowedException;
import com.xurxodev.stringcalculator.exception.ValidatorException;

public class NegativeNumberValidateRule implements NumberValidatorByValidateRules.ValidateRule {

    @Override
    public void validate(Integer number) throws ValidatorException {
        if (number < 0)
            throw new NegativeNumbersNotAllowedException();
    }
}
